GitHubTeams.ps1

```diff
@@ -276,6 +313,484 @@ filter Get-GitHubTeamMember
+        The name of the team's organization.
+
+    .PARAMETER TeamName
+        The name of the team.
+
+    .PARAMETER Description
+        The description for the team.
+
+    .PARAMETER Privacy
+        The level of privacy this team should have.
+
+    .PARAMETER ParentTeamName
+        The name of a team to set as the parent team.
+
+    .PARAMETER AccessToken
+        If provided, this will be used as the AccessToken for authentication with the
+        REST Api.  Otherwise, will attempt to use the configured value or will run unauthenticated.
+
+    .PARAMETER NoStatus
+        If this switch is specified, long-running commands will run on the main thread
+        with no commandline status update.  When not specified, those commands run in
+        the background, enabling the command prompt to provide status information.
+        If not supplied here, the DefaultNoStatus configuration property value will be used.
+
+    .INPUTS
+        GitHub.Team
+        System.String
+
+    .OUTPUTS
+        GitHub.User
+
+    .EXAMPLE
+        Set-GitHubTeam -OrganizationName PowerShell -TeamName Developers -Description 'New Description'
+
+        Updates the description for the 'Developers' GitHub team in the 'PowerShell' organization.
+
+    .EXAMPLE
+        $team = Get-GitHubTeam -OrganizationName PowerShell -TeamName Developers
+        $team | Set-GitHubTeam -Description 'New Description'
+
+        You can also pipe in a GitHub team that was returned from a previous command.
+#>
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSShouldProcess', '',
+        Justification = 'Methods called within here make use of PSShouldProcess, and the switch
+        is passed on to them inherently.')]
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSReviewUnusedParameter', '',
+        Justification = 'One or more parameters (like NoStatus) are only referenced by helper
+        methods which get access to it from the stack via Get-Variable -Scope 1.')]
+    [CmdletBinding(
+        SupportsShouldProcess,
+        PositionalBinding = $false
+    )]
+    [OutputType({$script:GitHubTeamTypeName})]
+    param
+    (
+        [Parameter(
+            Mandatory,
+            ValueFromPipelineByPropertyName,
+            Position = 1)]
+        [ValidateNotNullOrEmpty()]
+        [string] $OrganizationName,
+
+        [Parameter(
+            Mandatory,
+            ValueFromPipelineByPropertyName,
+            Position = 2)]
+        [ValidateNotNullOrEmpty()]
+        [string] $TeamName,
+
+        [string] $Description,
+
+        [ValidateSet('Secret','Closed')]
+        [string] $Privacy,
+
+        [string] $ParentTeamName,
+
+        [string] $AccessToken,
+
+        [switch] $NoStatus
+    )
+
+    Write-InvocationLog
+
+    $telemetryProperties = @{
+        OrganizationName = (Get-PiiSafeString -PlainText $OrganizationName)
+        TeamName = (Get-PiiSafeString -PlainText $TeamName)
+    }
+
+    $getGitHubTeamParms = @{
```

Comment:
Similar to New/Remove, there should be a parameterset that doesn't require there to be an ancillary `Get` call just to complete the normal case.  That means that the `TeamSlug` needs to be an alternative parameterset to `TeamName` and it should support pipeline input by property name so that it can be piped in.

Reply:
I don't intend to implement slugs in this PR.

Reply:
It is certainly your prerogative to stand by that, however if you don't implement slugs in this PR, be prepared for the function signature to change in a follow-up PR (either by me or someone else) in order to fix this functionality gap.

Reply:
So just to clarify the conclusion to these conversations, I I'm not intending to implement slug support in this PR as I have no interest in the functionality. In my view it can be added later by someone else through another PR as a non-breaking change.

Reply:
sounds good.  I'll handle the slug update once this goes in.  And your feedback around the status handling from a different thread has been heard (and I agree with you), and my current thoughts are captured in #247.  I'd be interested in your thoughts there.